tests/queries/tests.py

```diff
@@ -1382,6 +1383,21 @@ def test_combine_join_reuse(self):
         self.assertEqual(len(combined), 1)
         self.assertEqual(combined[0].name, 'a1')
 
+    def test_ticket26522(self):
+        # This should not non-deterministically raise an AssertionError because
+        # change_map contains a circular reference. Reuse join aliases in-order.
```

Comment:
Join aliases are reused in order. This shouldn't non-deterministically raise AssertionError because
change_map contains a circular reference (#26522).